apps/nextdns/nextdns.star

```diff
@@ -0,0 +1,276 @@
+"""
+Applet: NextDNS
+Summary: NextDNS Account Stats
+Description: Displays NextDNS account total query & total blocked query counts + 7-day activity graph.
+Author: ndhotsky
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("http.star", "http")
+load("humanize.star", "humanize")
+load("math.star", "math")
+load("render.star", "render")
+load("schema.star", "schema")
+
+# Endpoints
+BASE_URL = "https://api.nextdns.io"
+TIME_SERIES_URL = ";series?"
+ANALYTICS_STATUS_ENDPOINT = "/profiles/{}/analytics/status"
+
+def build_get_request(endpoint, api_key, **kwargs):
+    """
+    Builds the URL and headers component of an HTTP GET request
+
+    Args:
+        endpoint: NextDNS endpoint
+        api_key: NextDNS account api key
+        **kwargs: optionally specify additional arggs to append to the url for time series requests
+
+    Returns:
+        URL string and headers dictionary containing NextDNS API key
+    """
+    suffix = ""
+
+    if kwargs.get("since"):
+        suffix = TIME_SERIES_URL + "from={}".format(kwargs["since"])
+        if kwargs.get("interval"):
+            suffix = suffix + "&interval={}".format(kwargs["interval"])
+        if kwargs.get("limit"):
+            suffix = suffix + "&limit={}".format(kwargs["limit"])
+
+    url = BASE_URL + endpoint + suffix
+    headers = {"X-Api-Key": api_key}
+
+    return url, headers
+
+def query_nextdns(api_key, profile_id, endpoint, **kwargs):
+    """
+    Queries NextDNS' API with caching enabled
+
+    Args:
+        api_key: NextDNS API key
+        profile_id: NextDNS profile id
+        endpoint: NextDNS API endpoint
+        **kwargs: Used to specify NextDNS API url parameters when applicable
+
+    Returns:
+        New or cached JSON response from NextDNS
+    """
+    endpoint = endpoint.format(profile_id)
+    url, headers = build_get_request(endpoint, api_key, since = kwargs.get("since", None), interval = kwargs.get("interval", None), limit = kwargs.get("limit", None))
+
+    cached_data = cache.get(url)
```

Comment:
Check out the new `ttl_seconds` parameter on `http.get()`. It can simplify cache logic a fair bit.